### ISSUE TITLE
Rustdoc doesn't complain about ambiguous link

### ISSUE THREAD

**Issue by nashley** [rust-lang/rust#92544]
This causes `rustdoc` to link to `#method.foo`:

```rust
pub struct Test {
	pub foo: u8,
}

impl Test {
	/// Returns [Test::foo]
	pub fn foo(&self) -> u8 {
		self.foo
	}
}
```

This causes `rustdoc` to link to `#structfield.foo`:

```rust
pub struct Test {
	pub foo: u8,
}

impl Test {
	/// Returns [Test::foo]
	pub fn get_foo(&self) -> u8 {
		self.foo
	}
}
```

The only difference between the above two snippets is `Test::foo(&self)` being renamed to `Test::get_foo(&self)`.
For convenience, [here's a minimal repo with a commit demonstrating the issue](https://github.com/nashley/broken-doc-field-link/commits/main).

I expected to see `cargo doc` complain about an ambiguous link as described in [the RFC](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md#path-ambiguities).

Instead, `cargo doc` silently ignored the field with the same name and just linked to the method.

### Meta

`rustc --version --verbose`:
```
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-unknown-linux-gnu
release: 1.57.0
LLVM version: 13.0.0
```

The behavior is the same for `cargo +nightly doc`.
`cargo +nightly rustc -- --version --verbose`:
```
rustc 1.59.0-nightly (8f3238f89 2022-01-02)
binary: rustc
commit-hash: 8f3238f898163f09726c3d2b2cc9bafb09da26f3
commit-date: 2022-01-02
host: x86_64-unknown-linux-gnu
release: 1.59.0-nightly
LLVM version: 13.0.0
```

There's no backtrace to include.

**Comment by nashley** [comment]
Additionally, I don't see a way to force `rustdoc` to link to the field name explicitly with `[Test.foo]` [as described in the RFC](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md#linking-to-fields), but perhaps that's expected.

**Comment by jyn514** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/80283.
